### PR TITLE
fix(http): correct JSON error position reporting

### LINKED ISSUE #5545: Incorrect character position for JSON syntax errors
### Brief summary

Response.json() reports an incorrect line/character position for JSON syntax errors (off by one).

### k6 version

v1.5.0

### OS

Windows 11

### Docker version and image (if applicable)

_No response_

### Steps to reproduce the problem

1. Start a local server that returns invalid JSON:
```
  python - <<'PY'
  from http.server import BaseHTTPRequestHandler, HTTPServer
  class H(BaseHTTPRequestHandler):
    def do_GET(self):
      self.send_response(200)
      self.send_header("Content-Type", "application/json")
      self.end_headers()
      self.wfile.write(b"abc")
  HTTPServer(("127.0.0.1", 8080), H).serve_forever()
  PY
```
2. script.js:
```
  import http from "k6/http";

  export const options = { vus: 1, iterations: 1 };

  export default function () {
    const res = http.get("http://127.0.0.1:8080/bad");
    res.json();
  }
```
3. Run:

`k6 run script.js`

### Expected behaviour

The error message should report line 1, character 1 because the invalid character is the very first byte.

### Actual behaviour

The error message reports line 1, character 2, e.g.

### PR DESCRIPTION
**Pull request by chojs23**
Normalize json.SyntaxError.Offset (1-based byte count) before computing line/character and stop counting once the offset is reached.

Closes #5545 

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests for my changes.
- [x] I have run linter and tests locally (`make check`) and all pass.

## Checklist: Documentation (only for k6 maintainers and if relevant)

**Please do not merge this PR until the following items are filled out.**

- [ ] I have added the correct milestone and labels to the PR.
- [ ] I have updated the release notes: _link_
- [ ] I have updated or added an issue to the [k6-documentation](https://github.com/grafana/k6-docs): grafana/k6-docs#NUMBER if applicable
- [ ] I have updated or added an issue to the [TypeScript definitions](https://github.com/grafana/k6-DefinitelyTyped/tree/master/types/k6): grafana/k6-DefinitelyTyped#NUMBER if applicable

### CODE REVIEW

File: js/modules/k6/http/response.go [hunk 1]
```diff
@@ -118,15 +118,25 @@ func checkErrorInJSON(input []byte, offset int, err error) error {
...
 	for i, b := range str {
+		if i > offset {
+			break
+		}
 		if b == lf {
 			line++
 			character = 0
+			continue
 		}
 		character++
```
Comment (ankur22) [inline js/modules/k6/http/response.go]: Would it work the same if the `if i > offset` check was moved after `character++`? Then we might not need `if character == 0` outside the for loop.

```go
character++
if i > offset {
    break
}
```

WDYT?

Comment (chojs23) [reply js/modules/k6/http/response.go]: @ankur22 Thanks! With the current condition `i > offset`, moving the break after `character++` changes behavior: for `offset == 1` (normalized to 0), `character` becomes 2 instead of 1. 
So it’s not equivalent as-is. If we want to simplify, we can switch to `i >= offset` and handle the newline case (set `character = 1` when the offset is exactly at `\n`), which preserves the current outputs and removes the trailing `if character == 0` block.

Comment (ankur22) [reply js/modules/k6/http/response.go]: Ah! Good point. Any chance you could create some more unit tests to cover the edge cases you've mentioned? I'm happy with the implementation as it tbh.

Comment (chojs23) [reply js/modules/k6/http/response.go]: @ankur22 
I modified it as you suggested (this looks better). I also added the unit test for newline-at-offset behavior.

File: js/modules/k6/http/response_test.go [hunk 2]
```diff
@@ -104,6 +105,15 @@ func invalidJSONHandler(w http.ResponseWriter, _ *http.Request) {
...
 }
 
+func TestCheckErrorInJSON_NewlineOffset(t *testing.T) {
+	err := checkErrorInJSON([]byte("a\nb"), 2, errors.New("boom"))
+
+	var jsonErr jsonError
+	require.ErrorAs(t, err, &jsonErr)
+	assert.Equal(t, 2, jsonErr.line)
+	assert.Equal(t, 1, jsonErr.character)
+}
```
Comment (ankur22) [inline js/modules/k6/http/response_test.go]: Thanks for the test. This doesn't look correct, because the actual Go json unmarshaller would point to the first byte in this case. I'm wondering if we can turn this into a table test which works with the real json unmarshaller, or expanding the TestResponse_Json test with extra edge cases.

Comment (chojs23) [reply js/modules/k6/http/response_test.go]: @ankur22 Nice catch! Thanks

File: js/modules/k6/http/response_test.go [hunk 3]
```diff
@@ -104,6 +104,46 @@ func invalidJSONHandler(w http.ResponseWriter, _ *http.Request) {
...
+			name:     "invalid-after-newline",
+			input:    "true\nx",
+			wantLine: 2,
+			wantChar: 1,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			var v interface{}
```
Comment (inancgumus) [inline js/modules/k6/http/response_test.go]: Please use `any`:

```suggestion
			var v any
```

File: js/modules/k6/http/response_test.go [hunk 4]
```diff
@@ -104,6 +104,46 @@ func invalidJSONHandler(w http.ResponseWriter, _ *http.Request) {
...
+			input:    "a\nb",
+			wantLine: 1,
+			wantChar: 1,
+		},
+		{
+			name:     "invalid-after-newline",
+			input:    "true\nx",
+			wantLine: 2,
+			wantChar: 1,
+		},
```
Comment (inancgumus) [inline js/modules/k6/http/response_test.go:125]: Can you add another test case with a valid JSON but a JSON syntax error inside (not just offsets, but rather in the middle)?

Comment (chojs23) [reply js/modules/k6/http/response_test.go:125]: @inancgumus sure, Thanks!

**Review by inancgumus — APPROVED** [review]
Thanks for the contribution ❤️ 🚀
